assets/helpers/payPalExpressCheckout/payPalExpressCheckout.js

```diff
@@ -52,57 +51,61 @@ function payPalRequestData(bodyObj: Object, csrfToken: string) {
   };
 }
 
-function setupPayment(dispatch: Function, state: CombinedState) {
-
-  const payPalState = state.payPalExpressCheckout;
-  const csrfToken = state.csrf.token;
-
-  return (resolve, reject) => {
-
-    const requestBody = {
-      amount: payPalState.amount,
-      billingPeriod: payPalState.billingPeriod,
-      currency: payPalState.currency,
-    };
-
-    fetch(routes.payPalSetupPayment, payPalRequestData(requestBody, csrfToken || ''))
-      .then(handleSetupResponse)
-      .then((token) => {
-        if (token) {
-          resolve(token.token);
-        } else {
-          dispatch(payPalExpressError('PayPal token came back blank.'));
-        }
-      }).catch((err) => {
-        dispatch(payPalExpressError(err.message));
-        reject(err);
-      });
+function setupPayment(
+  amount: number,
+  billingPeriod: string,
+  currency: IsoCurrency,
+  csrfToken: string,
+) {
+
+  const requestBody = {
+    amount,
+    billingPeriod,
+    currency,
   };
+
+  return fetch(routes.payPalSetupPayment, payPalRequestData(requestBody, csrfToken))
+    .then(handleSetupResponse)
+    .then((token) => {
+      if (token) {
+        return token.token;
+      }
+      throw new Error('Invalid token');
+    });
 }
 
-function createAgreement(payPalData: Object, state: CombinedState) {
+function createAgreement(payPalData: Object, csrfToken: string) {
   const body = { token: payPalData.paymentToken };
-  const csrfToken = state.csrf.token;
 
-  return fetch(routes.payPalCreateAgreement, payPalRequestData(body, csrfToken || ''))
+  return fetch(routes.payPalCreateAgreement, payPalRequestData(body, csrfToken))
     .then(response => response.json());
 }
 
-function setup(dispatch: Function, getState: () => CombinedState, callback: Function) {
+function setup(
```

Comment:
Is there any way of breaking this into smaller functions? At 80 lines it's a bit difficult to parse.

Reply:
Good idea.